components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3200,6 +3415,58 @@ else if (APIConstants.API_KEY_TYPE_SANDBOX.equals(tokenType)) { // if
         return keyDetails;
     }
 
+    @Override
+    public Map<String, String> completeApplicationRegistration(String userId, int applicationId, String tokenType, String tokenScope, String groupingId) throws APIManagementException {
```

Comment:
Is this a newly added method?

Reply:
Yes. This is a newly added method to support application registration using application ID